### PR TITLE
[STRATCONN-3087] Pass statsContext to audience methods

### PR DESCRIPTION
**Pull request by marinhero**
Pass statsContext object to create/get audience methods so the `statsClient` is available from within. This will be very useful for the future migrations we will run.

## Testing

Testing completed successfully via localhost. I only verified the availability of the object and not its use. I'll do that with the upcoming Tiktok createAudience method.

Screenshot 2023-08-17 at 3 42 01 PM

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by nick-Ag — APPROVED** [review]

File: packages/core/src/destination-kit/index.ts [hunk 1]
```diff
@@ -92,12 +96,14 @@ export interface AudienceDestinationConfigurationWithCreateGet<Settings = unknow
   extends AudienceDestinationConfiguration {
   createAudience(
     request: RequestClient,
-    createAudienceInput: CreateAudienceInput<Settings, AudienceSettings>
+    createAudienceInput: CreateAudienceInput<Settings, AudienceSettings>,
+    statsContext: StatsContext | undefined
```
[the remark this answers is not in the document]
Comment (SyedWasiHaider) [reply packages/core/src/destination-kit/index.ts:100]: in other words, couldn't one just do `createAudienceInput.statsContext` ?

[the remark this answers is not in the document]
Comment (marinhero) [reply packages/core/src/destination-kit/index.ts]: Good callout, I've removed it.

**Comment by marinhero** [comment]
Test Evidence available in the DD Dashboard.: Screenshot 2023-08-23 at 1 00 38 PM

**Review by SyedWasiHaider — APPROVED** [review]

**Comment by marinhero** [comment]
Hey @joe-ayoub-segment FYI this is an update to the core package. Please bump it when you perform the weekly deploy. Thanks!

**Comment by joe-ayoub-segment** [comment]
hi @marinhero this PR has been deployed

**Comment by joe-ayoub-segment** [comment]
> Hey @joe-ayoub-segment FYI this is an update to the core package. Please bump it when you perform the weekly deploy. Thanks!

Hi @marinhero , can you provide some step by step instructions for what I need to do please?

**Pull request merged**